noir-projects/noir-protocol-circuits/crates/types/src/utils/arrays.nr

```diff
@@ -142,6 +142,19 @@ where
     result
 }
 
+// Helper fn to create a subarray from a given array
+pub fn array_splice<T, let N: u32, let M: u32>(array: [T; N], offset: u32) -> [T; M]
+where
+    T: Empty,
+{
+    assert(M + offset <= N, "Subarray length larger than array length");
```

Comment:
I'm guessing the offset can't be determined at compile time, when you call this function?

Reply:
Looking at the gate counts, I think it is determining it at compile time when I input a constant. I left `offset` as an input so the function could be used with a constant or variable input and be more useful in the long run.

Reply:
Oh that's clever!